nano/secure/blockstore.cpp

```diff
@@ -3,14 +3,54 @@
 
 #include <boost/endian/conversion.hpp>
 
-nano::block_sideband::block_sideband (nano::block_type type_a, nano::account const & account_a, nano::block_hash const & successor_a, nano::amount const & balance_a, uint64_t height_a, uint64_t timestamp_a, nano::epoch epoch_a) :
+uint8_t nano::block_details::packed () const
+{
+	uint8_t result{ 0 };
+	result += static_cast<uint8_t> (is_send) << 7;
+	result += static_cast<uint8_t> (is_receive) << 6;
+	result += static_cast<uint8_t> (is_epoch) << 5;
+	result += (static_cast<uint8_t> (static_cast<uint8_t> (epoch) << 3)) >> 3;
+	return result;
+}
+
+void nano::block_details::unpack (uint8_t details_a)
+{
+	is_send = static_cast<bool> (details_a >> 7);
+	is_receive = static_cast<bool> ((static_cast<uint8_t> (details_a << 1)) >> 7);
+	is_epoch = static_cast<bool> (static_cast<uint8_t> (details_a << 2) >> 7);
+	epoch = static_cast<nano::epoch> (static_cast<uint8_t> (details_a << 3) >> 3);
+}
```

Comment:
Can using a `std::bitset<8>` instead of a `uint8_t` make this code a bit simpler? e.g can do:
```
is_send = bits.test (7);
is_receive = bits.test (6);
// etc
```

https://en.cppreference.com/w/cpp/utility/bitset

Reply:
Thanks! Means converting to bitset and then to ulong but makes it less prone to errors (which I did initially).